File: Java/Sorting Algorithms/Selection Sort/SelectionSort.java
class SelectionSort {

    public static void selectionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[i]) {// find smaller number
                    // swap elements
                    int temp = arr[j];
                    arr[j] = arr[i];
                    arr[i] = temp;
                }
            }
        }
    }

    public static void main(String args[]) {
        int[] arr = { 8, 1, 4, 10, 12, 5, 2, 3, 7 };
        System.out.println("Array before Selection Sort");

        for (int i : arr) {
            System.out.print(i + " ");
        }

        selectionSort(arr);// sorting array elements using selection sort

        System.out.println("\nArray after Selection Sort");
        for (int i : arr) {
            System.out.print(i + " ");
        }
    }
}

File: golang/Sorting Algorithms/Selection Sort/selectionSort.go
package main

import "fmt"

func main() {
	arr := []int{3, 8, 11, 12, 5, 4, 5, 2, 1, 10}

	fmt.Println("Array before Selection Sort")
	fmt.Println(arr)
	
	selectionSort(arr)
	
	fmt.Println("Array after Selection Sort")
	fmt.Println(arr)
}

func selectionSort(arr []int) {
	for i := 0; i < len(arr) - 1; i++ {
		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[i] {
				arr[j], arr[i] = arr[i], arr[j] //Swap elements
			}
		}
	}
}
